gui/symfony-docker/src/Service/BlockChainService.php

```diff
@@ -135,4 +165,40 @@ public function getStringDataFromTokenID(int $tokenId): array
         return $returnData;
     }
 
+    public function getPossibleResourceFromResourceID(int $resourceId, String $role, String $resourceType): array
```

Comment:
I don't understand this function is for, can you briefly explain please. 

You get templates of a role, then keep only the one with needed_resources==resourceID  AND resource_type==resourceType --> So you have here the templates you only can create with the parameters. 

Be careful, you're only checking the first occurrence of the needed_resources array, there may be more than one.

confusing for me.. need explication please

Reply:
//return all the possible resource from a certain resourceID, role and resourceType
    //for example, if we want to get all the possible from a certain resourceID,
    //this function will return all the possible resource from this resourceID

Reply:
what its meaning ? all the possible CHILD or PARENT resource ?

Reply:
With this function in theory you get a list of all "Child" resource you can make with :
sheep => sheep carcass 
sheep carcass  => sheep demi carcass 
sheep carcass  => list of all meat you can make with sheep carcass
...

Reply:
good to know, thx